### PR TITLE
[tests-only] Adjust unskipped file tests

### PR DESCRIPTION
**Pull request by saw-jan**
part of https://github.com/owncloud/owncloud-sdk/issues/644 and https://github.com/owncloud/owncloud-sdk/issues/780

Adjust pact interaction for unskipped tests in filesTest.js

### CODE REVIEW

File: tests/filesTest.js [hunk 1]
```diff
@@ -59,15 +66,21 @@ describe('Main: Currently testing files management,', function () {
           dMultistatus.setAttributes({ 'xmlns:d': 'DAV:' })
           dMultistatus
             .appendElement('d:response', '', dResponse => {
-              dResponse.appendElement('d:href', '', `/remote.php/webdav/${parentFolder}/`)
+              dResponse.appendElement(
+                'd:href', '',
+                MatchersV3.regex(
+                  `.*\\/remote\\.php\\/webdav\\/${parentFolder}\\/`,
```
Comment (dpakach) [inline tests/filesTest.js:72]:
```suggestion
                  `\\/remote\\.php\\/webdav\\/${parentFolder}\\/`,
```

File: tests/filesTest.js [hunk 2]
```diff
@@ -100,14 +121,26 @@ describe('Main: Currently testing files management,', function () {
     for (const subFile of items) {
       response.push(node => {
         node
-          .appendElement('d:href', '', `/remote.php/webdav/${testFolder}/${subFile}`)
+          .appendElement(
+            'd:href', '',
+            MatchersV3.regex(
+              `*.\\/remote\\.php\\/webdav\\/${testFolder}\\/${subFile}`,
```
Comment (dpakach) [inline tests/filesTest.js:127]:
```suggestion
              `\\/remote\\.php\\/webdav\\/${testFolder}\\/${subFile}`,
```

File: tests/filesTest.js [hunk 3]
```diff
@@ -149,45 +184,13 @@ describe('Main: Currently testing files management,', function () {
           dMultistatus.setAttributes({ 'xmlns:d': 'DAV:', 'xmlns:s': 'http://sabredav.org/ns', 'xmlns:oc': 'http://owncloud.org/ns' })
           dMultistatus
             .appendElement('d:response', '', dResponse => {
-              dResponse.appendElement('d:href', '', `/remote.php/webdav/${testFolder}/${testFile}`)
+              dResponse.appendElement('d:href', '',
+                MatchersV3.regex(`.*\\/remote\\.php\\/webdav\\/${file}`, `/remote.php/webdav/${file}`)
```
Comment (dpakach) [inline tests/filesTest.js:188]:
```suggestion
                MatchersV3.regex(`\\/remote\\.php\\/webdav\\/${file}`, `/remote.php/webdav/${file}`)
```

File: tests/filesTest.js [hunk 4]
```diff
@@ -821,10 +924,15 @@ describe('Main: Currently testing files management,', function () {
               'xmlns:oc': 'http://owncloud.org/ns'
             })
             dMultistatus.appendElement('d:response', '', dResponse => {
-              dResponse.appendElement('d:href', '', '/remote.php/dav/meta/123456789/')
+              dResponse.appendElement('d:href', '',
+                MatchersV3.regex(
+                  `.*\\/remote\\.php\\/webdav\\/${file}`,
```
Comment (dpakach) [inline tests/filesTest.js:929]:
```suggestion
                  `\\/remote\\.php\\/webdav\\/${file}`,
```

File: tests/filesTest.js [hunk 5]
```diff
@@ -1413,29 +1500,19 @@ describe('Main: Currently testing files management,', function () {
...
-        return oc.files.fileInfo(newFile, ['{http://owncloud.org/ns}fileid'])
-          .then(fileInfo => {
-            fileId = fileInfo.getFileId()
-            return oc.systemTags.createTag({ name: newTagName })
-          }).then(resp => {
-            tagId = resp
-            return oc.systemTags.tagFile(fileId, tagId)
-          }).then(() => {
-            return oc.files.getFilesByTags([tagId], ['{http://owncloud.org/ns}fileid'])
-          }).then(files => {
```
Comment (dpakach) [inline tests/filesTest.js:1438]: tests are fileInfo and getFileID are already present, but I dont see any test for createTags, You need to add that. Or you can leave that test as it is and refactor this later

Comment (saw-jan) [reply tests/filesTest.js:1438]: done in PR https://github.com/owncloud/owncloud-sdk/pull/792
waiting that to be merged

File: tests/filesTest.js (hunk 4 above)
Comment (saw-jan) [reply tests/filesTest.js:929]: why is that?
if we run server from sub folder then href will be something like `/sub/remote.php/webdav/....`

Comment (dpakach) [reply tests/filesTest.js:929]: oh yes it does, I didn't considered that. sorry about that.

**Review by dpakach — COMMENTED** [review]
lgtm :+1:

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

**Review by individual-it — CHANGES_REQUESTED** [review]
a lot of tests that I would expect to pass on oc10 are still marked as failing on oc10

**Comment by saw-jan** [comment]
> a lot of tests that I would expect to pass on oc10 are still marked as failing on oc10

@individual-it  Could you point out some for me?
I need to check them

File: tests/filesTest.js [hunk 6]
```diff
@@ -1398,115 +1541,58 @@ describe('Main: Currently testing files management,', function () {
...
-          .then(fileInfo => {
-            expect(fileInfo.getProperty('{http://owncloud.org/ns}fileid')).toEqual('123456789')
-          }).catch(error => {
-            expect(error).toBe(null)
-          })
-      })
-    })
-
     it('checking method: tag file', async function () {
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1567]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 7]
```diff
@@ -1381,11 +1521,14 @@ describe('Main: Currently testing files management,', function () {
     })
 
     it('checking method: create tag', async function () {
-      const newTagName = 'testSystemTag12345'
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1524]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 8]
```diff
@@ -1337,39 +1474,42 @@ describe('Main: Currently testing files management,', function () {
         }
       }
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1476]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 9]
```diff
@@ -1242,16 +1361,22 @@ describe('Main: Currently testing files management,', function () {
       ]
 
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1363]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 10]
```diff
@@ -1171,16 +1285,20 @@ describe('Main: Currently testing files management,', function () {
 
     it('checking method: favorite filter', async function () {
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1287]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 11]
```diff
@@ -1146,23 +1260,23 @@ describe('Main: Currently testing files management,', function () {
   describe('favorite, search file', function () {
     const fileId = 123456789
     const tagId = 6789
+    const file = `${testFolder}/${testFile}`
+    const newTagName = 'testSystemTag12345'
 
     it('checking method: favorite', async function () {
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1267]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

File: tests/filesTest.js [hunk 12]
```diff
@@ -1122,20 +1237,19 @@ describe('Main: Currently testing files management,', function () {
   describe('unfavorite a file', function () {
     it('checking method: unfavorite', async function () {
       const provider = createProvider(true, true)
```
Comment (individual-it) [inline tests/filesTest.js:1239]: should pass now in oc10, can you set it to `const provider = createProvider(false, true)`

**Comment by individual-it** [comment]
@saw-jan see new comments, or are those blocked by https://github.com/pact-foundation/pact-js/issues/632 ?

**Comment by saw-jan** [comment]
Yeah, some of you mentioned are blocked. But `create tag` and `tag file` should pass

**Review by individual-it — APPROVED** [review]

**Pull request merged**